src/wp-includes/media-template.php

```diff
@@ -357,8 +357,23 @@ function wp_print_media_templates() {
 			<div class="thumbnail thumbnail-{{ data.type }}">
 				<# if ( data.uploading ) { #>
 					<div class="media-progress-bar"><div></div></div>
-				<# } else if ( data.sizes && data.sizes.large ) { #>
-					<img class="details-image" src="{{ data.sizes.large.url }}" draggable="false" alt="" />
+				<# } else if ( data.sizes && data.sizes.large ) {
+					var large_image_preview_url = data.sizes.large.url;
+					if ( data.sizes.full ) {
+						var full_image_crop_id = data.sizes.full.url.match(/^.+-(e[0-9]{13})\.[^\\/]+$/)?.[1],
+							large_image_crop_id = data.sizes.large.url.match(/^.+-(e[0-9]{13})-[^\\/]+$/)?.[1];
+						<?php
+							// A cropped image should be used if the "large" image hasn't been updated during cropping.
+							// It can happen if the original image is larger than the cropped image and
+							// the cropped image is smaller than 1024x1024 pixels (the size of a "large" image).
```

Comment:
I suggest removing the default pixel dimensions (1024x1024) and only refer to the "Large size" as configured in *Settings > Media Settings*. (Through testing I confirmed that images smaller than the configured large dimensions were subject to the reported issue, so the exact pixel sizes may vary.)

Reply:
Thank you for the review, @ironprogrammer.
That's good point. Fixed in aa66d7cd22ae523fbe790ac720e6d0b70bb7577d